### PR TITLE
rubocup fixes

### PR DESCRIPTION
**Pull request by wfeldt**
fix a number of rubocup complaints and adjust config a bit

### CODE REVIEW

File: src/lib/storage/fake_device_factory.rb [hunk 1]
```diff
@@ -281,9 +287,9 @@ def create_partition(parent, args)
 
         # partitions are created in order, so first suitable slot should be fine
         # note: skip areas we marked as empty
-        slot = slots.find { |slot|
-          slot.possible?(type) && !@free_regions.member?(slot.region.start)
-        }
+        slot = slots.find do |s|
```
Comment (shundhammer) [inline src/lib/storage/fake_device_factory.rb]: I liked the old version better. Was that something Rubocop wanted to be changed? Or was it a line length limitation?

IHMO that only shows how absurd many of the Rubocop settings are.

**Comment by shundhammer** [comment]
Other than that one issue I commented above, LGTM.

File: src/lib/storage/fake_device_factory.rb (hunk 1 above)
Comment (wfeldt) [inline src/lib/storage/fake_device_factory.rb]: rubo wanted {} -> do ... end

Comment (shundhammer) [inline src/lib/storage/fake_device_factory.rb]: Well, yes, but then maybe simply keep "slot" as the variable name?

Comment (wfeldt) [inline src/lib/storage/fake_device_factory.rb]: Ah, you mean that. It wanted that also ('slot' shadows the outside 'slot').

**Comment by ancorgs** [comment]
#70 already solves all these offenses (and any other)

**Pull request closed without merging**